File: Garage_Rental.Infra/Service/CarService.cs
using Garage_Rental.Core.Data;
using Garage_Rental.Core.Repository;
using Garage_Rental.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;

namespace Garage_Rental.Infra.Service
{
    public class CarService : IGenericService<Car>
    {
        private readonly IGenericRepository<Car> _carRepository;

        public CarService(IGenericRepository<Car> CarRepository)
        {
            _carRepository = CarRepository;
        }

        public bool Create(Car car)
        {
            return _carRepository.Create(car);
        }

        public void Delete(int id)
        {
            _carRepository.Delete(id);
        }

        public List<Car> GetAll()
        {
            return _carRepository.GetAll();
        }

        public Car GetById(int id)
        {
            return _carRepository.GetById(id);
        }

        public void Update(Car car)
        {
            _carRepository.Update(car);
        }
    }
}

File: FrontEnd/GarageRentalFrontEnd/src/app/Services/car.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Car } from '../Models/car.model';

@Injectable({
  providedIn: 'root'
})
export class CarService {


 baseApiUrl:string = environment.baseApiUrl;

  constructor(private http:HttpClient) { }


  getCarList(): Observable<Car[]>{
    return this.http.get<Car[]>(this.baseApiUrl +'Car/GetAll');
  }

  getCar(id:string):Observable<Car>{
    return this.http.get<Car>(this.baseApiUrl +'/api/Car/GetById?id=' + id);
  }

  addCar(addCarReq:Car):Observable<Car>{
    return this.http.post<Car>(this.baseApiUrl +'/api/Car/Create',addCarReq);
  }

  updateCar(id:number,updateCarreq:Car):Observable<Car>{
    return this.http.post<Car>(this.baseApiUrl +'/api/Car/Update?id='+ id ,updateCarreq)
  }
  deleteCar(id:number):Observable<Car>{
    return this.http.delete<Car>(this.baseApiUrl +'/api/Car/Delete?id=' + id);
  }
}
